frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java

```diff
@@ -126,27 +135,20 @@ private void deleteIfRequiredAndRefresh(AnActionEvent anActionEvent, String bran
             .map(b -> b.equals(branchName))
             .getOrElse(true);
         if (slidOutBranchIsCurrent) {
-          LOG.warn("Skipping local branch deletion because it is equal to current branch");
+          LOG.debug("Skipping local branch deletion because it is equal to current branch");
+          return;
         }
 
-        java.util.Map<String, java.util.List<? extends GitRepository>> branchesToContainingRepositories = java.util.Collections
-            .singletonMap(branchName, selectedVcsRepository.toJavaList());
-        GitBrancher.getInstance(project).deleteBranches(branchesToContainingRepositories,
-            () -> getGraphTable(anActionEvent).queueRepositoryUpdateAndModelRefresh());
-        isRefreshScheduled = true;
+        GitBrancher.getInstance(project).deleteBranch(branchName, selectedVcsRepository.toJavaList());
       }
     }
-
-    if (!isRefreshScheduled) {
-      getGraphTable(anActionEvent).queueRepositoryUpdateAndModelRefresh();
-    }
   }
 
   private boolean getDeleteLocalBranchOnSlideOutGitConfigKeyValue(VirtualFile root, Project project) {
```

Comment:
Until Checker Framework allows us to express this more systematically (`@Heavyweight` or so)... pls add a javadoc stating that this method must NOT be called on UI thread (and same for `deleteBranchIfRequired`)